### DISCUSSION TITLE
一分钟读论文：《自动将 Legacy 项目重构为 ES6》

### DISCUSSION THREAD

**Discussion by giscus** [unbug/unbug.github.io#7] [Announcements]
# 一分钟读论文：《自动将 Legacy 项目重构为 ES6》

很多遗留的项目都是 ES5 代码，不能在 ES6 或 TypeScript 下很好的复用，特别是企业级大型前端工程，维护成本令人头疼。雅典经济与商业大学信息学系的论文《Automated Refactoring of Legacy JavaScript Code to ES6 Modules》实现了将遗留 ES5 代码自动重构为 ES6 模块，重点是将模块对象解构为多个可单独重用的模块来实现模块内容的细粒度重用，每个重构模块支持按需引入而不是引入整个模块对象，并自动重建依赖关系，最大限度减少耦合并提高了可维护性。

https://unbug.github.io/Automated-Refactoring-of-Legacy-JavaScript-Code-to-ES6-Modules/

**Comment by unbug** [comment]
完整的ES6 语法、重建模块依赖关系、模块支持按需引入，全都做到了还是很赞的👍
